### ISSUE TITLE
Rails Console Doesn't Appear to Honor Environment Argument

### ISSUE THREAD

**Issue by todd** [rails/rails#20039]
Noticed this while debugging a separate issue while upgrading one of our apps to Rails 4.2.1. Starting a Rails console in a non-development environment by specifying the environment as a command argument doesn't appear to load the correct configuration for that environment.

Here's a snippet from our `staging.rb`:
```ruby
Rails.application.configure do
  # ...  
  config.eager_load = true
  # ...
end
```

And here's some output from a staging environment Rails console:
```
$ bundle exec rails c staging
Loading staging environment (Rails 4.2.1)
[1] pry(main)> Rails.application.config.eager_load
=> false
```

However, specifying the environment through `ENV` works as expected.
```
$ RAILS_ENV=staging bundle exec rails c
Loading staging environment (Rails 4.2.1)
STAGING [1] pry(main)> Rails.application.config.eager_load
=> true
```

Is the former behavior no longer supported? I'd be willing to try to track the root of this down if it's a bug.

**Comment by todd** [comment]
I was actually unable to to reproduce this with a fresh app created from 4-2-stable and 4.2.1. There's most likely some weirdness with some library our app is using. I'm gonna close this for now and I'll report back here if I'm able to track down what's causing this in our app.

**Comment by todd** [comment]
I was able to trace this back to a patch we were applying early in the app's boot process. Essentially, we were referencing `Rails.env` before `RAILS_ENV` had had a chance to be set properly within Rails. This meant that running `rails c <env>` would result in the console loading the default development configuration. However, by setting `RAILS_ENV` from the shell, the proper configuration would be loaded as it wouldn't need to be overwritten later in the boot process. We worked around this by referencing `ENV['RAILS_ENV']` directly instead of using `Rails.env`.

Hope this helps anyone that encounters a similar problem.
